examples/reminders/lib/base/data_sources/local/realm_reminders_data_source.dart

```diff
@@ -0,0 +1,177 @@
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:realm/realm.dart';
+import 'package:uuid/uuid.dart' as uuid;
+
+import '../../models/reminder/reminder_list_response.dart';
+import '../../models/reminder/reminder_model.dart';
+import '../../models/reminder/reminder_realm_model.dart';
+import '../remote/reminders_data_source.dart';
+
+class RealmRemindersDataSource extends RemindersDataSource {
+  final Realm _realm;
+  final uuid.Uuid _uuid;
+  final FlutterSecureStorage _storage;
+
+  static const _authorId = 'authorId';
+  static const _anonymous = 'anonymous';
+
+  RealmRemindersDataSource(this._realm, this._uuid, this._storage);
+
+  @override
+  Future<ReminderModel> create(
+      {required String title,
+      required DateTime dueDate,
+      required bool complete}) async {
+    final authorId = await _getAuthorIdOrNull();
+    final reminder = ReminderRealmModel(
+      _uuid.v4(),
+      title,
+      dueDate,
+      complete,
+      authorId: authorId,
+    );
+    _realm.write(() {
```

Comment:
It's not an asynchronous operation, but since `RealmRemindersDataSource` is extending `RemindersDataSource` in order to retain minimal changes to the existing solution, this implementation is using Futures